libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -2570,6 +2603,8 @@ instance (Ord a, HasSpec fn a) => HasSpec fn (Set a) where
               genFromSpec elemS `suchThatT` (`Set.notMember` s)
         go (n - 1) (Set.insert e s)
 
+  shrinkWithTypeSpec (SetSpec _ es _) as = map Set.fromList $ shrinkList (shrinkWithSpec es) (Set.toList as)
```

Comment:
Wouldn't it make sense to also shrink the size spec here?

Reply:
The idea here is not to shrink the spec, only the values.